assets/data_table/main.js

```diff
@@ -0,0 +1,317 @@
+      return {
+        ...prevColSizes,
+        [column.title]: newSize,
+      };
+    });
+  }, []);
+
+  useEffect(() => {
+    ctx.handleEvent("update_content", (content) => {
+      setContent(content);
+    });
+  }, []);
+
+  useEffect(() => {
+    const icon = content.order === "asc" ? "arrowUp" : "arrowDown";
+    const newColumns = columns.map((header) => ({
+      ...header,
+      overlayIcon: header.id === content.order_by ? icon : null,
+    }));
+    setColumns(newColumns);
+  }, [content.order, content.order_by]);
+
+  useEffect(() => {
+    const newColumns = columns.map((header) => {
+      return { ...header, width: colSizes[header.title] };
+    });
+    setColumns(newColumns);
+  }, [colSizes]);
+
+  return (
+    <div className="app">
+      <div className="navigation">
+        <div className="navigation__info">
+          <h2 className="navigation__name">{data.name}</h2>
+          <span className="navigation__details">
+            {totalRows || "?"} entries
+          </span>
+        </div>
+        <div className="navigation__space"></div>
+        {hasRefetch && <Refetch />}
+        <Search toggleSearch={toggleSearch} />
+        <Limit limit={content.limit} totalRows={totalRows} ctx={ctx} />
+        {hasPagination && (
+          <Pagination
+            page={content.page}
+            maxPage={content.max_page}
+            ctx={ctx}
+          />
+        )}
+      </div>
+      {hasData && (
+        <DataEditor
+          className="table-container"
+          theme={theme}
+          getCellContent={getData}
+          columns={columns}
+          rows={rows}
+          width="100%"
+          height={height}
+          rowHeight={44}
+          headerHeight={44}
+          verticalBorder={false}
+          rowMarkers="both"
+          onHeaderClicked={orderBy}
+          showSearch={showSearch}
+          getCellsForSelection={true}
+          onSearchClose={toggleSearch}
+          headerIcons={customHeaderIcons}
+          overscrollX={100}
+          isDraggable={false}
+          smoothScrollX={true}
+          smoothScrollY={true}
+          onColumnResize={onColumnResize}
+        />
+      )}
+      {!hasData && <p className="no-data">No data</p>}
+      <div id="portal" />
+    </div>
+  );
+}
+
+function Refetch() {
+  const refetch = () => {
+    ctx.pushEvent("refetch");
+  };
+
+  return (
+    <button className="icon-button" aria-label="refresh" onClick={refetch}>
+      <RiRefreshLine />
+    </button>
+  );
+}
+
+function Search({ toggleSearch }) {
+  return (
+    <span className="tooltip right" data-tooltip="Current page search">
+      <button
+        className="icon-button search"
+        aria-label="search"
+        onClick={toggleSearch}
+      >
+        <RiSearch2Line />
+      </button>
+    </span>
+  );
+}
+
+function Limit({ limit, totalRows, ctx }) {
+  const handleChange = (event) => {
+    ctx.pushEvent("limit", { limit: parseInt(event.target.value) });
+  };
```

Comment:
I would pass `onChange` to the component and then invoke it with `parseInt(event.target.value)`. This way this component is less concerned with the actual operation.